File: bindings/dart/lib/src/syslog.dart
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Syslog protocol types for proven-servers.

/// Severity matching the Idris2 ABI tags.
enum Severity {
  emergency(0),
  severity_Alert(1),
  critical(2),
  error(3),
  warning(4),
  notice(5),
  informational(6),
  debug(7);

  const Severity(this.tag);
  final int tag;

  static Severity? fromTag(int tag) {
    for (final v in values) {
      if (v.tag == tag) return v;
    }
    return null;
  }
}

/// Facility matching the Idris2 ABI tags.
enum Facility {
  kern(0),
  user(1),
  mail(2),
  daemon(3),
  auth(4),
  syslog(5),
  lpr(6),
  news(7),
  uucp(8),
  cron(9),
  authPriv(10),
  ftp(11),
  ntp(12),
  audit(13),
  facility_Alert(14),
  clock(15),
  local0(16),
  local1(17),
  local2(18),
  local3(19),
  local4(20),
  local5(21),
  local6(22),
  local7(23);

  const Facility(this.tag);
  final int tag;

  static Facility? fromTag(int tag) {
    for (final v in values) {
      if (v.tag == tag) return v;
    }
    return null;
  }
}

/// Transport matching the Idris2 ABI tags.
enum Transport {
  udp514(0),
  tcp514(1),
  tls6514(2);

  const Transport(this.tag);
  final int tag;

  static Transport? fromTag(int tag) {
    for (final v in values) {
      if (v.tag == tag) return v;
    }
    return null;
  }
}

File: bindings/javascript/src/syslog.js
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Syslog protocol types for proven-servers.

/** Severity matching the Idris2 ABI tags. */
export const Severity = Object.freeze({
  EMERGENCY: 0,
  SEVERITY_ALERT: 1,
  CRITICAL: 2,
  ERROR: 3,
  WARNING: 4,
  NOTICE: 5,
  INFORMATIONAL: 6,
  DEBUG: 7,
});

/** Facility matching the Idris2 ABI tags. */
export const Facility = Object.freeze({
  KERN: 0,
  USER: 1,
  MAIL: 2,
  DAEMON: 3,
  AUTH: 4,
  SYSLOG: 5,
  LPR: 6,
  NEWS: 7,
  UUCP: 8,
  CRON: 9,
  AUTH_PRIV: 10,
  FTP: 11,
  NTP: 12,
  AUDIT: 13,
  FACILITY_ALERT: 14,
  CLOCK: 15,
  LOCAL0: 16,
  LOCAL1: 17,
  LOCAL2: 18,
  LOCAL3: 19,
  LOCAL4: 20,
  LOCAL5: 21,
  LOCAL6: 22,
  LOCAL7: 23,
});

/** Transport matching the Idris2 ABI tags. */
export const Transport = Object.freeze({
  UDP514: 0,
  TCP514: 1,
  TLS6514: 2,
});
